File: python/atelier_bench/exporters/pdf_exporter.py
"""pdf exporter."""

from __future__ import annotations

from collections.abc import Sequence

from ..contracts.report_contract import ReportRow
from ..support.abstract_exporter import AbstractExporter


class PdfExporter(AbstractExporter):
    def __init__(self) -> None:
        super().__init__("pdf", "application/pdf")

    def export(self, rows: Sequence[ReportRow]) -> str:
        return f"%PDF-1.7\n% {self.format_count(len(rows))} row(s)\n%%EOF"

File: go/atelier/exporters/pdf_exporter.go
// Package exporters carries the 8 exporters. Each embeds support.BaseExporter
// (Extension, FormatCell), declares only Export, and registers itself under
// its extension in an init function.
package exporters

import (
	"fmt"

	"atelier.example/lane/atelier/contracts"
	"atelier.example/lane/atelier/support"
)

// Compile-time proof of satisfaction.
var _ contracts.ExporterContract = (*PdfExporter)(nil)

// PdfExporter renders report rows as pdf.
type PdfExporter struct {
	support.BaseExporter
	MimeType string
}

// NewPdfExporter builds the exporter.
func NewPdfExporter() *PdfExporter {
	return &PdfExporter{BaseExporter: support.NewBaseExporter("pdf"), MimeType: "application/pdf"}
}

// Export is the only contract method this type declares.
func (e *PdfExporter) Export(rows []contracts.ReportRow) string {
	return fmt.Sprintf("%%PDF-1.7\n%% %d row(s)\n%%%%EOF", len(rows))
}

// init registers this exporter under its extension.
func init() {
	Register(NewPdfExporter())
}
